include/aws/common/hash_table.h

```diff
@@ -104,17 +103,19 @@ extern "C" {
     /**
      * Initializes a hash map with initial capacity for 'size' elements
      * without resizing. Uses hash_fn to compute the hash of each element.
-     * equals_fn to compute equality of two keys.  destroy_fn is called whenver
-     * an element is removed without being returned, and may be NULL if a
-     * callback is not desired in this case.
+     * equals_fn to compute equality of two keys.  Whenver an elements is
+     * removed without being returned, destroy_key_fn is run on the pointer
+     * to the key and destroy_value_fn is run on the pointer to the value.
+     * Either or both may be NULL if a callback is not desired in this case.
      */
 
     AWS_COMMON_API int aws_hash_table_init(
         struct aws_hash_table *map,
         struct aws_allocator *alloc, size_t size,
         aws_hash_fn_t hash_fn,
         aws_equals_fn_t equals_fn,
-        aws_hash_element_destroy_t destroy_fn);
+        aws_hash_element_destroy_t destroy_key_fn,
```

Comment:
While we're making changes, do we need a context parameter on the destroy fn? This may not be very useful in pure-C applications, but it might be useful when doing language bindings to pass a pointer to the foreign function to be invoked via a shim.

Reply:
I do not have any insight on this.